src/v2/models/AppState.js

```diff
@@ -107,6 +107,11 @@ export default Model.extend({
     }
 
     this.set(resp);
+
+    // response has messages object with the same form
+    if (!_.isEmpty(resp.currentState.messages)) {
+      this.trigger('showCallout', resp.currentState.messages);
```

Comment:
feels like we could consolidate `change:currentFormName` and `showCallout` to be something a little more generic like `idxResponseUpdated`

Reply:
In this case the formName actually does not change. We get back the same form with a `message` object in the response now.

Reply:
I meant we could use single event rather than `change:currentFormName` and `showCallout` to trigger FormController `render` upon API response. basically it says 'whenever we got API response and it's different from last time, re-render the view`

Reply:
Although this sounds good, I feel like this has a potential to cause regression in other places. I tried this change and I broke the switch factor flow. I think we should do this refactor later, possible after LEA.

Reply:
- now I realised how bad it is without unit test/e2e test in place..
- I think you need to fix `switchForm` in `FormControlller`. If still broken, create a jira for it.

Reply:
thats exactly why I am hesitant to make this change